shared/js/initApp.js

```diff
@@ -13,7 +13,8 @@ window.apps = {
   load: loadApp,
   // Legacy Blockly initialization that was moved here from _blockly.html.haml.
   // Modifies `appOptions` with some default values in `baseOptions`.
-  setup: function () {
+  // Move blockly-specific setup function out of shared and back into dashboard.
+  setupBlockly: function () {
```

Comment:
One thing we could do (which might be way to heavy handed) is to define the shape of appOptions somewhere, similar to how we have prop type validation. We would then warn any time we saw fields we didn't expect. Or if we wanted to be even more aggressive, we could remove fields we didn't explicitly allow. Another benefit of this is that we have somewhere to look to see the expected shape of the object.